e2e-tests/ukef/cypress/e2e/journeys/portal/deal-cancellation/gef/deal-cancellation-maker-unable-issue-facility.spec.js

```diff
@@ -0,0 +1,107 @@
+import MOCK_USERS from '../../../../../../../e2e-fixtures/portal-users.fixture';
+import { MOCK_APPLICATION_AIN_DRAFT, MOCK_APPLICATION_MIN_DRAFT } from '../../../../../../../e2e-fixtures/gef/mocks/mock-deals';
+import { anIssuedCashFacility, anUnissuedCashFacility, multipleMockGefFacilities } from '../../../../../../../e2e-fixtures/mock-gef-facilities';
+import { yesterday, tomorrow } from '../../../../../../../e2e-fixtures/dateConstants';
+
+const { BANK1_MAKER1 } = MOCK_USERS;
+
+context('Deal cancellation', () => {
+  const dealIdsWithEffectiveDate = [];
+  const dealFacilities = [];
+  const allDealFacilities = [];
+  const ainDeals = Array(4).fill(MOCK_APPLICATION_AIN_DRAFT);
+  const minDeals = Array(4).fill(MOCK_APPLICATION_MIN_DRAFT);
+  const gefDeals = [...ainDeals, ...minDeals];
+
+  const facilities = [
+    anIssuedCashFacility({ facilityEndDateEnabled: true }),
+    anIssuedCashFacility({ facilityEndDateEnabled: true }),
+    anUnissuedCashFacility({ facilityEndDateEnabled: true }),
+    multipleMockGefFacilities({ facilityEndDateEnabled: true }).unissuedContingentFacility,
+    anIssuedCashFacility({ facilityEndDateEnabled: true }),
+    anIssuedCashFacility({ facilityEndDateEnabled: true }),
+    multipleMockGefFacilities({ facilityEndDateEnabled: true }).unissuedContingentFacility,
+    anUnissuedCashFacility({ facilityEndDateEnabled: true }),
+  ];
+
+  before(() => {
+    cy.insertManyGefDeals(gefDeals, BANK1_MAKER1).then((insertedDeals) => {
+      insertedDeals.forEach((deal, index) => {
+        dealIdsWithEffectiveDate.push({ id: deal._id, effectiveDate: index % 2 === 0 ? tomorrow.date : yesterday.date });
+        // updates a gef deal so has relevant fields
+        cy.updateGefDeal(deal._id, MOCK_APPLICATION_AIN_DRAFT, BANK1_MAKER1);
+
+        cy.createGefFacilities(deal._id, [facilities[index]], BANK1_MAKER1).then((createdFacilities) => {
+          dealFacilities.push(createdFacilities.details);
+        });
+        allDealFacilities.push(dealFacilities);
+      });
+    });
+  });
+
+  beforeEach(() => {
+    cy.clearCookie('dtfs-session');
+    cy.clearCookie('_csrf');
+    cy.getCookies().should('be.empty');
+  });
+
+  after(() => {
+    cy.clearCookies();
+    cy.clearCookie('dtfs-session');
+    cy.clearCookie('_csrf');
+    cy.getCookies().should('be.empty');
+  });
+
+  describe('Deal cancellations on tfm with effective dates in the past and in the future', () => {
+    before(() => {
+      dealIdsWithEffectiveDate.forEach((dealIdWithEffectiveDate) => {
+        cy.gefDealCancellationFlow(dealIdWithEffectiveDate.id, dealIdWithEffectiveDate.effectiveDate);
+      });
+      cy.clearCookie('dtfs-session');
+      cy.clearCookie('_csrf');
+      cy.getCookies().should('be.empty');
+      cy.login(BANK1_MAKER1);
+    });
+
+    it('Gef AIN deal with issued facilities is submitted to UKEF, user cancel deal in the future in TFM. Maker unable to issue facility on portal', () => {
```

Comment:
It feels a bit weird that whether this test needs the effective date to be in the future, but its just arbitrarily accessing the 1st entry in the array. This feels like a great chance to use a builder - but that might be overkill!

I'd prefer something like this at least, so that the order of insertions isnt affecting our tests
`const deal = deals.filter(({ effectiveDate, submissionType  }) => effectiveDate == today.date && submissionType === 'AIN' )[]`